clang/docs/StandardCPlusPlusModules.rst

```diff
@@ -484,6 +511,13 @@ fragment is disabled by default. These checks can be enabled by specifying
 and you encounter incorrect or missing diagnostics, please report them via the
 `community issue tracker <https://github.com/llvm/llvm-project/issues/>`_.
 
+Privacy Issue
+-------------
+
+BMIs are not and should not be treated as an information hiding mechanism.
+They should always be assumed to contain all the information that was used to
+create them, in a recoverable form.
```

Comment:
I might expand on this a bit with something like, "While the `-fmodules_embed_all_files` flag explicitly emits the source code into the BMI file, the contents of the BMI file contain a sufficiently verbose(word choice?) representation to reproduce the original source file."

I'd like to make this warning as clear as possible: If you ship a BMI, you mind as well just ship the source code.

Reply:
Done by adding the suggested text to the above section about ``-fmodules-embed-all-files``. I feel it is odd to mention ``-fmodules-embed-all-files`` in the current section since it is mostly about clarifying that hiding information is not a goal for BMIs. It is true with and without ``-fmodules-embed-all-files``.